source/common/vb_gui.c

```diff
@@ -418,14 +446,12 @@ int emulation_sstate(void) {
     return D_EXIT;
 }
 
-int emulation_lstate(void) {
+int emulation_lstate(int state) {
     int i;
     int ret;
     int id,ver,crc;
-    char sspath[131];
     FILE* state_file;
-
-    sprintf(sspath, "%s.rds", tVBOpt.ROM_NAME);
+    char* sspath = get_savestate_path(state);
```

Comment:
When will the memory be released?

Reply:
addressed in [1561bf](https://github.com/Piiman/red-viper/commit/18561bffe44680dbee11116bd3e8412ae1ec0f1a)